src/write.rs

```diff
@@ -337,13 +337,20 @@ where
                 // we can drain out the operations that only the w_handle copy needs
                 //
                 // NOTE: the if above is because drain(0..0) would remove 0
-                for op in self.oplog.drain(0..self.swap_index) {
+                for op in self
+                    .oplog
+                    .drain(0..self.swap_index)
+                    .map(|op| op.unwrap_or_else(|| unreachable!("Nones are always temporary")))
```

Comment:
Why `.unwrap_or_else(|| unreachable` here instead of just `.expect`?

Also, I think I'd prefer to do the unwrap/expect inside the loop instead of making the iterator this unwieldy.

Reply:
When I started learning Rust I remember hearing something about `.unwrap_or_else(|| unreachable!())` being more efficient and have just rolled with it since. Really thinking about it now, yeah, it really shouldn't be. Will change all occurrences.

Reply:
Done: f74cadd688a8542f34d8e4797b37ae1c9cb920ae